snow/engine/snowman/transitive.go

```diff
@@ -478,41 +483,79 @@ func (t *Transitive) Start(ctx context.Context, startReqID uint32) error {
 		return err
 	}
 
+	var (
+		preferredID    ids.ID
+		preferredBlock snowman.Block
+	)
+
+	// Default to using the last accepted block if the VM does not define its
+	// own initial preferred block.
+	if t.getInitialPreferenceVM == nil {
+		preferredID = lastAcceptedID
+		preferredBlock = lastAccepted
+	} else {
+		preferredID, err = t.getInitialPreferenceVM.GetInitialPreference(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get initial preference: %w", err)
+		}
+
+		preferredBlock, err = t.VM.GetBlock(ctx, preferredID)
+		if err != nil {
+			return fmt.Errorf("failed to get initial preferred block: %w", err)
+		}
+	}
+
+	// Check if the initial setup is more recent than the last accepted
+	// block. This can occur if we shut down while a set of blocks was still
+	// processing in consensus.
+	preferredTip := lastAcceptedID
+	if preferredBlock.Height() > lastAccepted.Height() {
+		preferredTip = preferredID
+	}
```

Comment:
I don't think this handles the case correctly that the persisted preferences have been rejected by bootstrapping.

Reply:
Updated + added test